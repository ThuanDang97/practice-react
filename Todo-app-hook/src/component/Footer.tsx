import { memo } from 'react';
import Button from './common/Button';
import { filterBtns } from '../constants/MockData';

type TProps = {
  filterTodo: (value: string) => void;
  lengthTodo: number;
};

const Footer = (props: TProps): JSX.Element => {
  const { filterTodo, lengthTodo } = props;

  return (
    <footer className="footer">
      <span className="todo-count">
        <strong>{lengthTodo}</strong>
        <span> items </span>
        <span> left </span>
      </span>
      <ul className="filters">
        {filterBtns.map((btn) => (
          <li key={btn.id}>
            <Button {...btn} onUpdateFilterValue={filterTodo} />
          </li>
        ))}
      </ul>
    </footer>
  );
};

export default memo(Footer);
