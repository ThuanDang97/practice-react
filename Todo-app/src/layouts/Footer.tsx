import Button from '../component/Button';
import React, { Component } from 'react';
import { filterBtns } from '../constants/MockData';

type Props = {
  filterTodo: Function;
  lengthTodo: number;
};
class Footer extends Component<Props> {
  constructor(props: Props) {
    super(props);
  }
  render(): JSX.Element {
    return (
      <footer className="footer">
        <span className="todo-count">
          <strong>{this.props.lengthTodo}</strong>
          <span> items </span>
          <span> left </span>
        </span>
        <ul className="filters">
          {filterBtns.map((btn) => (
            <Button key={btn.id} {...btn} onUpdateFilterValue={this.props.filterTodo} />
          ))}
        </ul>
      </footer>
    );
  }
}

export default Footer;
