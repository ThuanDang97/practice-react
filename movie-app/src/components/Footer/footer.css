footer {
  max-width: 100%;
  width: 100vw;
  box-sizing: border-box;
  padding: 0;
  background-image: radial-gradient(at 30% top, #031d33 0%, rgba(3, 37, 65, 1) 70%);
  display: flex;
  align-content: space-between;
  justify-content: center;
}

.logo_footer {
  margin-right: 40px;
  width: 154px;
  height: 20px;
  margin-top: 20%;
}

.menu_footer {
  display: flex;
  justify-content: space-between;
  padding-top: 80px;
  padding-bottom: 80px;
  gap: 40px;
}

.nav_footer {
  display: flex;
  flex-direction: column;
  gap: 7px;
}

.nav_title {
  color: white;
  font-weight: 800;
  font-size: 24px;
  padding: 0;
  margin: 0;
}

.sub_nav {
  display: flex;
  flex-direction: column;
  gap: 7px;
}

.sub_nav_item {
  color: white;
}
