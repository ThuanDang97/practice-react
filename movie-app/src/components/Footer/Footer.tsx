import { Link } from 'react-router-dom';
import './footer.css';
import logo from '../../assets/logo-footer.svg';
import { navFooter } from '../../constants/MockData';
import { INavbar } from '../../interfaces/Navbar/INav';
import { IItem } from '../../interfaces/Navbar/IItem';

const Footer = (): JSX.Element => {
  const renderNavFooter = (itemNav: INavbar[]): JSX.Element => {
    return (
      <div className="menu_footer">
        {itemNav.map((item) => (
          <div key={item.id} className="nav_footer">
            <h3 className="nav_title">{item.title}</h3>
            <div className="sub_nav">{renderSubNavFooter(item.items)}</div>
          </div>
        ))}
      </div>
    );
  };

  const renderSubNavFooter = (itemSubNav: IItem[]) => {
    return (
      <>
        {itemSubNav.map((item) => (
          <Link to={item.pathItem} key={item.idItem} className="sub_nav_item">
            <span>{item.titleItem}</span>
          </Link>
        ))}
      </>
    );
  };

  return (
    <footer>
      <Link to="/">
        <h2 className="logo_footer">
          <img src={logo} alt="" />
        </h2>
      </Link>
      {renderNavFooter(navFooter)}
    </footer>
  );
};

export default Footer;
