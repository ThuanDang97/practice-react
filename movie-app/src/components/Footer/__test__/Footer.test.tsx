import { render } from '@testing-library/react';
import Footer from '../Footer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Test Footer component', () => {
  test('Component Footer matches DOM Snapshot', () => {
    const component = (
      <Router>
        <Footer />
      </Router>
    );
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });
});
