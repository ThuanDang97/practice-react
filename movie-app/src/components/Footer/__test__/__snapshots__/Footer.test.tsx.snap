// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer component Component Footer matches DOM Snapshot 1`] = `
<div>
  <footer>
    <a
      href="/"
    >
      <h2
        class="logo_footer"
      >
        <img
          alt=""
          src="IMAGE_MOCK"
        />
      </h2>
    </a>
    <div
      class="menu_footer"
    >
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          THE BASICS
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              About TMDB
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Contact Us
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Support Forums
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              API
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              System Status
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          GET INVOLVED
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Contribution Bible
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Add New Movie
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Add New TV Show
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          COMMUNITY
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Guidelines
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Discussions
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Leaderboard
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Twitter
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          LEGAL
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Terms of Use
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              API Terms of Use
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Privacy Policy
            </span>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
